lib/install/management/store_files.go

```diff
@@ -53,15 +53,37 @@ func (d *Dispatcher) DeleteStores(vchVM *vm.VirtualMachine, conf *metadata.Virtu
 	var emptyImages bool
 	var emptyVolumes bool
 	log.Infof("Removing images")
-	if emptyImages, err = d.deleteImages(ds, p); err != nil {
-		errs = append(errs, err.Error())
+	for _, imageDir := range conf.ImageStores {
+		imageDSes, err := d.session.Finder.DatastoreList(d.ctx, imageDir.Host)
+		if err != nil {
+			errs = append(errs, err.Error())
+			continue
+		}
+
+		if len(imageDSes) != 1 {
+			errs = append(errs, fmt.Sprintf("Invalid or ambiguous datastore name %s provided while attempting to remove image stores", imageDir.Host))
+			continue
+		}
+
+		if emptyImages, err = d.deleteParent(imageDSes[0], imageDir.Path); err != nil {
+			errs = append(errs, err.Error())
+		}
+
+		if !emptyImages {
+			log.Infof("Not deleting [%s] %s as it still contains files after removing images", imageDir.Host, imageDir.Path)
+			continue
+		}
+
+		if _, err = d.deleteParent(imageDSes[0], imageDir.Path); err != nil {
```

Comment:
As you already tried to delete everything under image store path, you should not call deleteParent here again, it will generate delete failure error message for file not found.